adapters/sspbc/sspbc.go

```diff
@@ -0,0 +1,407 @@
+	SiteId      string `json:"siteid"`
+	SlotId      string `json:"slotid"`
+}
+
+type SspbcAdapter struct {
+	version  string
+	endpoint string
+	// adslots mapping
+	// map key is slot id (as sent and received from proxy)
+	adSlots        map[string]AdSlotData
+	adSizes        map[string]int
+	bannerTemplate *template.Template
+}
+
+// ---------------ADAPTER INTERFACE------------------
+// Builder builds a new instance of the sspBC adapter
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	// find path to template file
+	pathToTemplate := "./adapters/sspbc/bannerTemplate.html"
+	workingDir, err := os.Getwd()
+	if err == nil && strings.HasSuffix(workingDir, "sspbc") {
+		// this is a test running in adapter's directory
+		pathToTemplate = "bannerTemplate.html"
+	} else if err != nil {
+		// this error does not break createBannerAd flow
+		glog.Errorf("SSPBC: Cannot get working directory, assuming default path")
+	}
+
+	bannerTemplate, err := template.ParseFiles(pathToTemplate)
+	if err != nil {
+		return nil, err
+	}
+
+	bidder := &SspbcAdapter{
+		endpoint:       config.Endpoint,
+		version:        version,
+		bannerTemplate: bannerTemplate,
+	}
+
+	return bidder, nil
+}
+
+func (a *SspbcAdapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errors []error
+
+	formattedRequest, err := formatSsbcRequest(a, request)
+	if err != nil {
+		glog.Errorf("SSPBC: cannot prepare request")
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	requestJSON, err := json.Marshal(formattedRequest)
+	if err != nil {
+		glog.Errorf("SSPBC: cannot marshal request")
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	requestData := &adapters.RequestData{
+		Method: "POST",
+		Uri:    fmt.Sprintf("%s?bdver=%s&inver=0", a.endpoint, a.version),
+		Body:   requestJSON,
+	}
+
+	return []*adapters.RequestData{requestData}, nil
+}
+
+func (a *SspbcAdapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, externalResponse *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	/*
+		  proxy responds with the following format
+			{
+			"cur": "PLN",
+			"id": "...",
+			"seatbid": [
+				{
+					"bid": [
+						{
+							"adm": "....",
+							"adomain": [
+								"sspbc-test"
+							],
+							"crid": "1234",
+							"ext": {
+								"adlabel": "Reklama",
+								"pubid": "431",
+								"siteid": "237503",
+								"slotid": "005",
+								"tagid": "slot"
+							},
+							"h": 250,
+						"id": "...",
+								"impid": "005",
+								"price": 95.95,
+							"w": 300
+						}
+					],
+					"seat": "sspbc-test"
+				}
+			],
+			"sn": "sspbc-test"
+			}
+
+		Note - we cannot read site SN, since response.sn is not defined in
+		openRTB2.BidResponse structure
+
+		For now we set SN as sspbc_go
+
+		Long term SN should be returned in bid.ext
+	*/
```

Comment:
I like the detail of the comment, but consider moving it to a markdown file in the directory to separate it from the code flow.

Reply:
OK,  I've removed unnecessary info from comments.